test/Common/Helpers/AssertHelper.cs

```diff
@@ -84,6 +87,116 @@ public static JsonElement AssertJsonProperty(JsonElement element, string propert
+                    Assert.Equal(JsonValueKind.Null, b.ValueKind);
+                    break;
+                default:
+                    throw new XunitException($"Bad JsonValueKind '{a.ValueKind}'");
+            }
+        }
+
+        private static void AssertEqualJsonObject(JsonElement a, JsonElement b)
+        {
+            Debug.Assert(a.ValueKind == JsonValueKind.Object && b.ValueKind == JsonValueKind.Object);
+
+            var aObjectEnumerator = a.EnumerateObject();
+            var bObjectEnumerator = b.EnumerateObject();
+
+            while (true)
+            {
+                var aCanMove = aObjectEnumerator.MoveNext();
+                var bCanMove = bObjectEnumerator.MoveNext();
+
+                if (aCanMove)
+                {
+                    Assert.True(bCanMove, $"a was able to enumerate over object '{a}' but b was NOT able to '{b}'");
+                }
+                else
+                {
+                    Assert.False(bCanMove, $"a was NOT able to enumerate over object '{a}' but b was able to '{b}'");
+                }
+
+                if (aCanMove == false && bCanMove == false)
+                {
+                    // They both can't continue to enumerate at the same time, that is valid
+                    break;
+                }
+
+                var aProp = aObjectEnumerator.Current;
+                var bProp = bObjectEnumerator.Current;
+
+                Assert.Equal(aProp.Name, bProp.Name);
+                // Recursion!
+                AssertEqualJson(aProp.Value, bProp.Value);
+            }
+        }
+
+        private static void AssertEqualJsonArray(JsonElement a, JsonElement b)
+        {
+            Debug.Assert(a.ValueKind == JsonValueKind.Array && b.ValueKind == JsonValueKind.Array);
+
+            var aArrayEnumerator = a.EnumerateArray();
+            var bArrayEnumerator = b.EnumerateArray();
+
+            while (true)
+            {
+                var aCanMove = aArrayEnumerator.MoveNext();
+                var bCanMove = bArrayEnumerator.MoveNext();
+
+                if (aCanMove)
+                {
+                    Assert.True(bCanMove, $"a was able to enumerate over array '{a}' but b was NOT able to '{b}'");
+                }
+                else
+                {
+                    Assert.False(bCanMove, $"a was NOT able to enumerate over array '{a}' but b was able to '{b}'");
+                }
+
+                if (aCanMove == false && bCanMove == false)
+                {
+                    // They both can't continue to enumerate at the same time, that is valid
+                    break;
+                }
+
+                var aElement = aArrayEnumerator.Current;
+                var bElement = bArrayEnumerator.Current;
+
+                // Recursion!
+                AssertEqualJson(aElement, bElement);
+            }
+        }
+
+        public async static Task<T> AssertResponseTypeIsAsync<T>(HttpContext context)
```

Comment:
I know I keep nitpicking this but can we take away the `Async`? Readability will be so much easier

Reply:
Done